es-rest-client-1.3/core/src/main/java/com/bazaarvoice/elasticsearch/client/core/util/ExceptionUtil.java

```diff
@@ -0,0 +1,47 @@
+package com.bazaarvoice.elasticsearch.client.core.util;
+
+import org.elasticsearch.index.Index;
+import org.elasticsearch.indices.IndexMissingException;
+
+import java.util.Map;
+import java.util.Objects;
+
+public final class ExceptionUtil {
+
+    private ExceptionUtil() {
+    }
+
+    /**
+     * Gets the appropriate exception from a result map from a RESTful call. The map is
+     * expected to have an entry for 'error' that described the error that occurred.
+     *
+     * Currently only supports special handling for IndexMissingExceptions. If the
+     * cause is an IndexMissingException then that will be returned otherwise the
+     * error message will be wrapped with a RuntimeException and returned.
+     *
+     * @param map non-null result map from RESTful call
+     * @return proper exception; IndexMissingException or RuntimeException with error message
+     *          in the 'error' map entry.
+     */
+    public static RuntimeException getProperException(Map<String, Object> map) {
+
+        RuntimeException ex = new RuntimeException("Encountered error: " + Objects.toString(map));
```

Comment:
Matter of style. I don't like having multiple return statements in as if else if block. They do the same thing. I will be consistent though and change it.